bip-v3.mediawiki

```diff
@@ -0,0 +1,254 @@
+
+There are weaknesses with this approach of using 2 anchors and CPFP Carve Out. This proposal helps address a few of them (see Related Work for how other weaknesses are addressed):
+
+* A switch to cluster mempool necessitates the removal of CPFP Carve Out <ref>https://delvingbitcoin.org/t/an-overview-of-the-cluster-mempool-proposal/393#the-cpfp-carveout-rule-can-no-longer-be-supported-12</ref>.
+* CPFP Carve Out only allows *one more* child to be added to the transaction. This means it cannot guarantee the ability to CPFP for more than 2 parties of a shared transaction.
+
+==Specification==
+
+This document describes one set of policy rules that can realistically be deployed today and is useful to today's applications. If mempool improvements enable more accurate rules or new application requirements emerge, it may be appropriate to implement a different set of policy rules to achieve the same goal.
+<ref>Examples of potential changes:
+* If increasing the maximum child size enables a new use case while still providing acceptable pinning bounds, we can loosen that rule.
+* If [https://delvingbitcoin.org/t/an-overview-of-the-cluster-mempool-proposal/393 cluster mempool] is implemented and some of the rules could be applied to all transactions instead of just v3, there is no need to specify them as additional rules.
+* The implementation of things like "replacement must be more incentive compatible" could change.
+</ref>
+
+===1 Parent with 1 Small Child===
+
+In addition to the node's other standardness and policy rules, the following rules apply to transactions with <code>nVersion</code> field set to 3 ("v3 transaction").
+
+1. A v3 transaction signals replaceability, even if it does not signal BIP125 replaceability.
+
+2. Any descendant of an unconfirmed v3 transaction must also be v3.
+<ref>Rationale:
+
+Combined with Rule 1, this gives us the property of "inherited signaling" when descendants of unconfirmed transactions are created. Additionally, checking whether a transaction signals replaceability this way does not require mempool traversal, and does not change based on what transactions are mined.
+</ref>
+Note: A v3 transaction can spend outputs from _confirmed_ non-v3 transactions.
+
+3. A v3 transaction's unconfirmed ancestors must all be v3.
+<ref>Rationale:
+
+Ensure the ancestor feerate rule does not underestimate a to-be-replaced v3 mempool transaction's incentive compatibility. Imagine the original transaction, A, has a child B and co-parent C (i.e. B spends from A and C). C also has another child, D. B is one of the original transactions and thus its ancestor feerate must be lower than the package's. However, this may be an underestimation because D can bump C without B's help. This is resolved if v3 transactions can only have v3 ancestors, as then C cannot have another child.
```

Comment:
The last sentence makes use of Rule 4 which is only defined below. Perhaps the order of these two rules should be swapped?

Reply:
Consolidated the rules a bit to be more understandable and not have these referencing problems.